### DISCUSSION TITLE
best-practices/simplify-nixos-related-commands

### DISCUSSION THREAD

**Discussion by giscus** [ryan4yin/nixos-and-flakes-book#197] [Announcements]
# best-practices/simplify-nixos-related-commands

An unofficial and opinionated book for beginners

https://nixos-and-flakes.thiscute.world/best-practices/simplify-nixos-related-commands

**Comment by AntonOfTheWoods** [comment]
These just seem to be shell aliases (or a function for `upp`). Are there some other advantages I'm missing with the above list?

**Reply by ryan4yin** [reply]
Just is a project-level command runner. You typically stick a Justfile in your project repository and manage it with Git. Just only works its magic within the project folder and can handle some pretty complex command orchestrations.

If your needs are straightforward, using shell aliases is totally fine. However, shell aliases are a bit more basic in functionality, and they can clutter up your entire shell session.
